src/librustc_errors/lib.rs

```diff
@@ -23,16 +23,9 @@
 #![feature(staged_api)]
 #![feature(range_contains)]
 #![feature(libc)]
-#![feature(unicode)]
 
-extern crate serialize;
 extern crate term;
-#[macro_use]
-extern crate log;
-#[macro_use]
 extern crate libc;
-extern crate std_unicode;
-extern crate serialize as rustc_serialize; // used by deriving
```

Comment:
And all these unused crates will still live undetected in `Cargo.toml`s and introduce build dependencies :(
`rustc` needs to somehow communicate this information to `cargo` so it could warn too.
cc @alexcrichton